undi/undi_analysis.py

```diff
@@ -0,0 +1,284 @@
+import sys, json, argparse
+from itertools import product
+import numpy as np
+
+from ase import Atoms
+from ase.io import read
+from ase.neighborlist import neighbor_list
+
+from undi import MuonNuclearInteraction
+from .isotopes import Element
+
+angtom = 1e-10
+
+def gen_neighbouring_atomic_structure(atoms, isotopes, spins, hdim_max):
+
+    ai,aj,d,D = neighbor_list('ijdD',atoms, 10.) # very large cutoff to get all possible interactions.
+                                              # Actual selection is done below
+
+    try:
+        EFGs = atoms.get_array('efg') # EFG should be stored in the atoms object. Done via read_magres_file() function (see below).
+    except:
+        EFGs = np.array([0])
+        
+    # sort
+    srt = np.argsort(d)
+    ai = ai[srt]
+    aj = aj[srt]
+    d = d[srt]
+    D = D[srt]
+
+    data = []
+    muon_pos = np.array([0,0,0])
```

Comment:
does this means that we need to put always the muon in the origin? or we can parse the muon_pos from the ASE object and assign muon_pos = atoms.get_positions()[-1]?

Reply:
No, it's fine, it is in the origin because we just calculate (vector) distances a few lines above.